### PR TITLE
Fix rare case in zlib decompresser

### PR DESCRIPTION
**Pull request by erikcorry**
Review in ignore-whitespace mode.

### CODE REVIEW

File: lib/zlib.toit [hunk 1]
```diff
@@ -704,29 +704,29 @@ class InflaterBackend implements Backend_:
...
-          add-symbol.call length-code
-        else:
-          last := 0
-          repeats := 3
-          if length-code == 16:
-            last = lengths_[counter_ - 1].bit-len
-            pending-bits_ = 2
-          else if length-code == 17:
-            pending-bits_ = 3
+        if counter_ != lengths_.size:
```
Comment (kasperl) [inline lib/zlib.toit:707]: Should this be an `else:` on the check in line 726?

File: tests/zlib-glitch-test.toit [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+// be found in the tests/LICENSE file.
+
+import zlib
+
+main:
+  decoder := zlib.BufferingInflater
+
+  decoder.write ZLIB_STREAM
+
+  while read := decoder.reader.read:
```
Comment (kasperl) [inline tests/zlib-glitch-test.toit:13]: Maybe explain how this failed before the fix?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: lib/zlib.toit (hunk 1 above)
Comment (erikcorry) [reply lib/zlib.toit:707]: No, add-symbol can update counter_.

**Pull request merged**